liferay-lima/liferay-portal
# LPS-151172 Make basic document type company scoped

# What is this about?

https://issues.liferay.com/browse/LPS-151172

The Basic Document Type was global to all companies (i.e. it wasstored under the SYSTEM company).  This goes against true multitenancyin the product.

This PR changes things so that there's one basic document type per company, and removes most usages of magic identifiers.

# How do I test it?

Follow the steps described in https://issues.liferay.com/browse/LPS-151172.

Additional tests will require running the upgrade process.  One possible scenario is the following:

1. Create a Collection to display basic document types.
2. Create two pages, each one with its own Asset Publisher portlet.
3. Configure one Asset Publisher to display the contents of the collection.
4. Configure the other Asset Publisher to display basic documents (legacy configuration).
5. Add a fragment to a page, and map the title of a basic document to it.

After running the upgrade, everything should be displayed correctly.
